### PR TITLE
Add minor stop 10281

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

File: OsmAnd-java/build.gradle [hunk 1]
```diff
@@ -112,6 +112,7 @@ dependencies {
 	implementation group: 'org.json', name: 'json', version: '20171018'
 	implementation 'it.unibo.alice.tuprolog:tuprolog:3.2.1'
 	implementation 'org.apache.commons:commons-compress:1.17'
```
Comment (vshcherb) [inline OsmAnd-java/build.gradle:114]: ради одного метода целую библиотеку? тем более в android?

File: OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java [hunk 2]
```diff
@@ -176,11 +176,43 @@ private boolean segmentLineBelongsToPolygon(CombineAreaRoutePoint p, CombineArea
 		combineWayPointsForAreaRouting(ctx, result);
 		validateAllPointsConnected(result);
 		splitRoadsAndAttachRoadSegments(ctx, result, recalculation);
+		for (int i = 0; i < result.size(); i++) {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java:179]: Name is quite bad - prepareStopSign

File: OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java [hunk 3]
```diff
@@ -176,11 +176,43 @@ private boolean segmentLineBelongsToPolygon(CombineAreaRoutePoint p, CombineArea
...
 		prepareTurnResults(ctx, result);
 		return result;
 	}
+	
+	public RouteSegmentResult prepareStopSign(RouteSegmentResult seg) {
+		int start = seg.getStartPointIndex();
+		int end = seg.getEndPointIndex();
+		List<Integer> stops = new ArrayList<>();
+
+		for (int i = start; i > end; i--) {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java:192]: this is totally incorrect! start is not > end

Comment (vshcherb) [reply OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java:192]: i == could be equal to start and to end point index, check other iterations

File: OsmAnd-java/build.gradle (hunk 1 above)
Comment (alisa911) [reply OsmAnd-java/build.gradle]: +

File: OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java (hunk 2 above)
Comment (alisa911) [reply OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java:179]: сделала filterMinorStops

File: OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java (hunk 3 above)
Comment (alisa911) [reply OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java]: исправила

File: OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java [hunk 4]
```diff
@@ -176,11 +176,54 @@ private boolean segmentLineBelongsToPolygon(CombineAreaRoutePoint p, CombineArea
...
+			start = startPoint;
+			end = endPoint;
+		} else {
+			start = endPoint;
+			end = startPoint;
+		}
+
+		List<Integer> stops = new ArrayList<>();
+
+		for (int i = start; i < end; i++) {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java:203]: there is much shorter form of this loop

Comment (vshcherb) [reply OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java:203]: is it correct that it doesn't check last point?

File: OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java [hunk 5]
```diff
@@ -176,11 +176,54 @@ private boolean segmentLineBelongsToPolygon(CombineAreaRoutePoint p, CombineArea
...
+
+		if (startPoint < endPoint) {
+			start = startPoint;
+			end = endPoint;
+		} else {
+			start = endPoint;
+			end = startPoint;
+		}
+
+		List<Integer> stops = new ArrayList<>();
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java:201]: Better not create object if it's not going to be used i.e. 
List<Integer> stops = null;
....
if(stops == null) {
... // create
}

if (stops != null) { 
// remove and check

}

**Pull request merged**